conans/test/generators/cmake_find_package_test.py

```diff
@@ -0,0 +1,125 @@
+import unittest
+from conans.test.utils.cpp_test_files import cpp_hello_conan_files
+from conans.test.utils.tools import TestClient
+from nose.plugins.attrib import attr
+
+
+@attr('slow')
+class CMakeFindPathGeneratorTest(unittest.TestCase):
+
+    def cmake_find_package_test(self):
+        """First package without custom find_package"""
+        client = TestClient()
+        files = cpp_hello_conan_files(name="Hello0",
+                                      settings='"os", "compiler", "arch", "build_type"')
+        client.save(files)
+        client.run("create . user/channel -s build_type=Release")
+
+        # Consume the previous Hello0 with auto generated FindHello0.cmake
+        # The module path will point to the "install" folder because of the cmake_paths.cmake
+        files = cpp_hello_conan_files(name="Hello1", deps=["Hello0/0.1@user/channel"],
+                                      settings='"os", "compiler", "arch", "build_type"')
+        files["conanfile.py"] = files["conanfile.py"].replace(
+            'generators = "cmake", "gcc"',
+            'generators = "cmake_paths", "cmake_find_package"')
+        files["CMakeLists.txt"] = """
+set(CMAKE_CXX_COMPILER_WORKS 1)
+set(CMAKE_CXX_ABI_COMPILED 1)
+project(MyHello CXX)
+cmake_minimum_required(VERSION 2.8)
+include("${CMAKE_BINARY_DIR}/conan_paths.cmake")
```

Comment:
I don't know if is there a Canonical way to do it. It is tested using both ways. 
About the `CMake` helper automatically injecting the toolchain, I think we shouldn't, because it is hard to know if the user already "patched" its CMakeLists to include it or not. But, for sure we have to document how to do it:

```
def build(self):
    cmake = CMake(self)
    cmake.definitions["CMAKE_TOOLCHAIN_FILE"] = "conan_paths.cmake"
    cmake.configure()
    cmake.build()
```
This generator is clearly focused on consumers, so the first documented way to do it (call it Canonical) should be with the toolchain as vcpkg. But we have to document other possibilities.

Reply:
What maybe we can do is to adjust the module path to the build folder in case `cmake_find_package` is detected. That is more similar to what pkg_config is doing. WDYT?